grammar.js

```diff
@@ -591,8 +618,7 @@ module.exports = grammar({
 
     call_expression: $ => prec(PREC.call, seq(
       field('function', $.expression),
-      field('arguments', $.arguments),
-      field('body', optional(choice($.block, $.case_block)))
+      field('arguments', choice($.arguments, $._case_clauses, $.block)),
```

Comment:
I get why this is necessary, but I was definitely hoping that _somehow_ we could get the `$._case_clauses` in

```scala
LIst(1,2,3).map {
  case x if x < 2 => "yes"
  case x => "no"
}
```

to be recognized as a `$.lambda_expression`

Reply:
hmm but that's not a lambda expression, it's essentially a block expression. From the Scala syntax summary:

```
  ArgumentExprs     ::=  ‘(’ [Exprs] ‘)’
                      |  ‘(’ [Exprs ‘,’] PostfixExpr ‘:’ ‘_’ ‘*’ ‘)’
                      |  [nl] BlockExpr
  BlockExpr         ::=  ‘{’ CaseClauses ‘}’
                      |  ‘{’ Block ‘}’```

Reply:
you're right. it's semantically an anonymous function, but lexically not! okay then.